cli/crates/server/src/bridge/server.rs

```diff
@@ -121,14 +138,98 @@ async fn search_endpoint(
     Ok(Json(response))
 }
 
+#[allow(clippy::too_many_lines)]
 async fn invoke_resolver_endpoint(
     State(handler_state): State<Arc<HandlerState>>,
     Json(payload): Json<ResolverInvocation>,
 ) -> Result<Json<serde_json::Value>, ApiError> {
     trace!("resolver invocation\n\n{:#?}\n", payload);
+
+    let environment = Environment::get();
+
+    let resolver_worker_port = loop {
+        let mut resolver_builds = handler_state.resolver_builds.lock().await;
+
+        let notify = if let Some(resolver_build) = resolver_builds.get(&payload.resolver_name) {
+            match resolver_build {
+                ResolverBuild::Succeeded { worker_port, .. } => break *worker_port,
+                ResolverBuild::Failed => return Err(ApiError::ResolverSpawnError),
+                ResolverBuild::InProgress { notify } => {
+                    if Arc::strong_count(notify) == 1 {
+                        let notify = notify.clone();
+                        resolver_builds.remove(&payload.resolver_name);
+                        notify
+                    } else {
+                        let notify = notify.clone();
+                        drop(resolver_builds);
+                        notify.notified().await;
+                        continue;
+                    }
+                }
+            }
+        } else {
+            Arc::new(Notify::new())
+        };
+
+        resolver_builds.insert(
```

Comment:
Because we also want to `insert()` in turn after the first `if` branch. 🙂 But I see what you mean, I guess we can not remove it in the first place (line 160). Fair enough, let me see…